tsc/meetings/2019-09-26.md

```diff
@@ -0,0 +1,112 @@
+Minutes from 28th OpenVDB TSC meeting, September 26th, 2019, (GMT)
+
+Attendees: *Nick* A., *Dan* B., *Jeff* L., *Ken* M., *Peter* C.
+
+Agenda:
+
+1) Confirm quorum
+2) Select TSC member to take minutes
+3) OpenVDB 6.2 postmortem
+4) Proposal for new improved interpolates
+5) (OVDB-117) MeshToVolume non-deterministic bug
+6) Windows CI
+7) Update on copyright notices
+8) User feedback (usd/python bindings/Maya support) 
+9) Plans for ABI=7
+
+1) Quorum was confirmed.
+
+2) Secretary was Nick Avramoussis
+
+3) OpenVDB 6.2 postmortem
+
+An issue was reported with the OpenVDB 6.2.0 release which broke ABI
+compatibility with Houdini. This was reproducible with a vanilla install of the
+toolkit (on Linux) and manifested during serialization of OpenVDB Point Data
+Grids into Houdini file formats (.bgeo and subsequent derivations) as a
+segmentation fault. Generally, the definition of ABI compatibility that we apply
+to the software is isolated to the Grid and Transform objects. Ken mentioned
+that the major ABI definition should guarantee a reinterpret cast of the Grid.
+However the io::StreamMetadata and internally held io::StreamMetadata::Impl
+objects can also be passed between libraries using OpenVDB. These classes are
+used for file IO and hold information about the current state of the IO stream.
+In the above case, the StreamMetadata object was being constructed by Houdini
+and passed across the library boundary to the writeBuffers() methods. This
+function is unique for OpenVDB Point serialization in that it accesses auxiliary
+stream metadata. Changes to the memory layout of the io::StreamMetadata::Impl
+class made in PR #436 (OVDB-91) meant that this data ended up incorrectly
+initialized when passed across the library boundary. Similar changes have been
+made to this class before, but have been fortunate enough that they only ever
+appended data members to the end of the class. This solves the issue so long as
```

Comment:
Data members have actually been _inserted_ into the class before. This may not have been an issue for one or both of these reasons - (1) the new data members were small enough in bytes so that structure padding and alignment resulted in the memory layout remaining the same, (2) data members after the inserted element(s) were not previously being accessed in the serialization routine.